From f7e7a221ff7eab809d6702084d1ae5a97806fe59 Mon Sep 17 00:00:00 2001
From: FakeNeth <cneth97@gmail.com>
Date: Thu, 9 Aug 2018 08:19:29 -0400
Subject: [PATCH] versioning fixes


diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index c45faf4..784f55e 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -1,10 +1,20 @@
 package org.bukkit.command.defaults;
 
+import com.destroystokyo.paper.VersionHistoryManager;
 import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
-
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantLock;
 import org.apache.commons.lang.Validate;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
@@ -12,26 +22,11 @@ import org.bukkit.command.CommandSender;
 import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.util.StringUtil;
-
-import com.google.common.collect.ImmutableList;
-import com.google.common.io.Resources;
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.net.URL;
-import java.net.URLEncoder;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.concurrent.locks.ReentrantLock;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
 
 // Paper start
-import java.io.InputStreamReader;
-import java.net.HttpURLConnection;
-import com.destroystokyo.paper.VersionHistoryManager;
 // Paper end
 
 public class VersionCommand extends BukkitCommand {
@@ -209,7 +204,27 @@ public class VersionCommand extends BukkitCommand {
     private void obtainVersion() {
         String version = Bukkit.getVersion();
         if (version == null) version = "Custom";
-        if (version.startsWith("git-Paper-")) {
+        // Scissors start
+        if (version.startsWith("git-Scissors-")) {
+            setVersionMessage("Scissors version information currently disabled.");
+//            String[] parts = version.substring("git-Scissors-".length()).split("[-\\s]");
+//            int distance = getDistance(null, parts[0]);
+//            switch (distance) {
+//                case -1:
+//                    setVersionMessage("Error obtaining version information");
+//                    break;
+//                case 0:
+//                    setVersionMessage("You are running the latest version");
+//                    break;
+//                case -2:
+//                    setVersionMessage("Unknown version");
+//                    break;
+//                default:
+//                    setVersionMessage("You are " + distance + " version(s) behind");
+//            }
+        }
+        else if (version.startsWith("git-Paper-")) {
+        // Scissors end
             String[] parts = version.substring("git-Paper-".length()).split("[-\\s]");
             int distance = getDistance(null, parts[0]);
             switch (distance) {
-- 
2.7.4

