From 22e59575961bf8ee1fafaa196876b65ee15fb93a Mon Sep 17 00:00:00 2001
From: FakeNeth <cneth97@gmail.com>
Date: Tue, 7 Aug 2018 14:16:17 -0400
Subject: [PATCH] brand server mod name


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 0399a48..1724725 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1155,7 +1155,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
     }
 
     public String getServerModName() {
-        return "Paper"; //Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "scissors"; // Scissors - scissors > //Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public CrashReport b(CrashReport crashreport) {
-- 
2.7.4

