From 25761b39d2e9f5d75a4dff64e84b13e5ad4c3d94 Mon Sep 17 00:00:00 2001
From: FakeNeth <cneth97@gmail.com>
Date: Tue, 7 Aug 2018 14:16:17 -0400
Subject: [PATCH] brand server mod name


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index e0546e3..02a3fad 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1156,7 +1156,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
     }
 
     public String getServerModName() {
-        return "Paper"; //Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "scissors"; // Scissors - scissors > //Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public CrashReport b(CrashReport crashreport) {
-- 
2.7.4

